src/main/scala/ai/nikin/typedgraph/core/Path.scala

```diff
@@ -0,0 +1,14 @@
+package ai.nikin.typedgraph.core
+
+class Path(val edges: List[Edge]) {
+  lazy val from:     Vertex       = edges.head.from
+  lazy val to:       Vertex       = edges.last.to
+  lazy val vertices: List[Vertex] = from :: edges.map(_.to)
+
+  def >>>(next: Vertex): Path = Path(edges, to >>> next)
+}
+
+object Path {
+  def apply(edges: Edge*): Path = new Path(edges.toList)
+  def apply(edges: List[Edge], moreEdges: Edge*): Path = new Path(edges ++ moreEdges)
```

Comment:
Isn't `def apply(edge: Edge, moreEdges: Edge*)` and removing `def apply(edges: Edge*)` more ergonomic? Otherwise you can end up with an empty List of Edges.

Reply:
done